engine/storage/snapshot/src/org/apache/cloudstack/storage/snapshot/StorageSystemSnapshotStrategy.java

```diff
@@ -459,12 +463,8 @@ public StrategyPriority canHandle(Snapshot snapshot, SnapshotOperation op) {
             storagePoolId = volumeVO.getPoolId();
         }
 
-        if (SnapshotOperation.REVERT.equals(op)) {
-            if (volumeVO != null && ImageFormat.QCOW2.equals(volumeVO.getFormat()))
-                return StrategyPriority.DEFAULT;
-            else
-                return StrategyPriority.CANT_HANDLE;
-        }
+        if (SnapshotOperation.REVERT.equals(op) && volumeVO != null && ImageFormat.QCOW2.equals(volumeVO.getFormat()))
```

Comment:
this can be moved to line 446. 
in case volumeVO is not null , the only thing between these lines is
storagePoolId = volumeVO.getPoolId();